src/ucp/core/ucp_listener.h

```diff
@@ -15,8 +15,19 @@
  */
 typedef struct ucp_listener {
     ucp_worker_h                   worker;
-    ucp_worker_iface_t             *wifaces;  /* Array of UCT interfaces to listen on */
-    uint8_t                        num_wifaces;
+
+    union {
+        struct {
+            ucp_worker_iface_t     *wifaces;  /* Array of UCT interfaces to listen on */
+            uint8_t                num_wifaces;
```

Comment:
maybe we can make the number of UCT elements a common field for both iface mode and CM mode: 
"ucp_rsc_index_t num_tls"